src/services/catalog.service.ts

```diff
@@ -0,0 +1,198 @@
+import * as angular from 'angular';
+import * as _ from 'lodash';
+
+export class CatalogService {
+  static $inject = ['Constants', '$filter'];
+
+  public $filter: any;
+  private constants: any;
+  private categories: any;
+
+  constructor(constants: any, $filter: any, $scope: any) {
+    this.constants = constants;
+    this.$filter = $filter;
+    this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+  }
+
+  public getServiceItem(resource: any) {
+    return new ServiceItem(resource, this);
+  }
+
+  public getImageItem(resource: any) {
+    return new ImageItem(resource, this);
+  }
+
+  public getCategoryFromSubCategory(subCategory: string) {
+    let retCategory: string = 'other';
+
+    let breakLoops: boolean = false;
+    _.each(this.categories, (category) => {
+      if (breakLoops) {
+        return false;
+      }
+      _.each(category.subCategories, (cat_subCategory) => {
+        if (cat_subCategory.id === subCategory) {
+          retCategory = category.id;
+          breakLoops = true;
+          return false;
+        }
+      });
+    });
+
+    return retCategory;
+  }
+
+  public getSubCategoryFromAlias(subCategory: string) {
+    let retSubCategory: string = subCategory;
+
+    let breakLoops: boolean = false;
+    _.each(this.categories, (category) => {
+      if (breakLoops) {
+        return false;
+      }
+      _.each(category.subCategories, (cat_subCategory) => {
+        if (cat_subCategory.categoryAliases) {
+          let result = _.findIndex(cat_subCategory.categoryAliases, (alias) => {
+            return alias === subCategory;
+          });
+          if (result !== -1) {
+            retSubCategory = cat_subCategory.id;
+            breakLoops = true;
+            return false;
+          }
+        }
+      });
+    });
+    return retSubCategory;
+  }
+}
+
+interface IServiceItem {
+  iconClass: string;
+  imageUrl: string;
+  name: string;
+  catsBySubCats: any;
+  description: string;
+  longDescription: string;
+  resource: any;
+}
+
+export class ServiceItem implements IServiceItem {
+  public iconClass: string;
+  public imageUrl: string;
+  public name: string;
+  public catsBySubCats: any;
+  public description: string;
+  public longDescription: string;
+  public resource: any;
+  private catalogSrv: CatalogService;
+
+  constructor (serviceClass: any, catalogSrv: CatalogService) {
+    this.resource = serviceClass;
+    this.catalogSrv = catalogSrv;
+    this.imageUrl = this.getImage();
```

Comment:
> I wonder if that's the right prefix if we want other brokers we write to
use the same annotation.  Maybe we should do something like
console.openshift.io since it's really a convention for the console?

makes sense.  i could also see broker.openshift.io/iconClass, though it is a bit redundant since anything providing that value is, by definition, a broker.